scarlet/spectrum.py

```diff
@@ -45,12 +45,12 @@ class TabulatedSpectrum(Spectrum):
         1D bounding box for focation of the spectrum in `frame`
     """
 
-    def __init__(self, frame, spectrum, bbox=None):
+    def __init__(self, frame, spectrum, bbox=None, step=1e-2):
```

Comment:
I'm not a fan of this option. The logic is the following: either you provide an array, which will be converted to a `Parameter` *with default everything*, or you provide your own `Parameter` instance. The middle ground is murky. Someone might want to change the constraint, but leave the step size etc. It's really hard to allow for some changes as options without changing others.

We could say, well, then we make all `Parameter` options (constraint, prior, step, even current gradient etc.) accessible to the user, but then you write interfaces that look a lot like a `Parameter` instantiation.

Reply:
I changed it by declaring the spectrum a Parameter object in the init_starlet_source